flink-cdc-connect/flink-cdc-pipeline-connectors/flink-connector-test-cdc/src/main/java/com/ververica/cdc/connectors/test/ValuesDatabase.java

```diff
@@ -0,0 +1,391 @@
+/*
+ * Copyright 2023 Ververica Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ververica.cdc.connectors.test;
+
+import com.ververica.cdc.common.data.RecordData;
+import com.ververica.cdc.common.event.AddColumnEvent;
+import com.ververica.cdc.common.event.AlterColumnTypeEvent;
+import com.ververica.cdc.common.event.CreateTableEvent;
+import com.ververica.cdc.common.event.DataChangeEvent;
+import com.ververica.cdc.common.event.DropColumnEvent;
+import com.ververica.cdc.common.event.RenameColumnEvent;
+import com.ververica.cdc.common.event.SchemaChangeEvent;
+import com.ververica.cdc.common.event.TableId;
+import com.ververica.cdc.common.schema.Column;
+import com.ververica.cdc.common.schema.Schema;
+import com.ververica.cdc.common.sink.MetadataApplier;
+import com.ververica.cdc.common.source.MetadataAccessor;
+import com.ververica.cdc.common.utils.Preconditions;
+import com.ververica.cdc.connectors.test.sink.ValuesDataSink;
+import com.ververica.cdc.connectors.test.source.ValuesDataSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+/**
+ * An auxiliary class for simulating a database, use for {@link ValuesDataSource} and {@link
+ * ValuesDataSink}.
+ */
+public class ValuesDatabase {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ValuesDatabase.class);
+
+    private static final Map<TableId, ValuesTable> globalTables = new ConcurrentHashMap<>();
+
+    /** use for {@link ValuesDataSource}. */
```

Comment:
Pleas add correct and meaningful comment, this is used be `ValuesDataSink` to apply upstream schema change event instead of `ValuesDataSoure`